op-challenger/game/fault/trace/select.go

```diff
@@ -0,0 +1,56 @@
+package trace
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ethereum-optimism/optimism/op-challenger/game/fault/types"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+func NewSimpleTraceAccessor(trace types.TraceProvider) *Accessor {
+	selector := func(_ context.Context, _ types.Game, _ types.Claim, _ types.Position) (types.TraceProvider, error) {
+		return trace, nil
+	}
+	return &Accessor{selector}
+}
+
+func NewSplitTraceAccessor(top types.TraceProvider, topDepth uint64, bottomFactory func(ctx context.Context, pre common.Hash, post common.Hash) (types.TraceProvider, error)) *Accessor {
```

Comment:
I'm tempted to move all the trace accessor stuff into the solver module - it exists because it requires too much knowledge of game details for a TraceProvider so it feels like it shouldn't be in the trace package either.  A key part of deciding that will be how it fits into the game registration though and I haven't actually played with that yet.